### DISCUSSION TITLE
How this project compares to gojq ?

### DISCUSSION THREAD

**Discussion by kpym** [mikefarah/yq#1021] [Q&A] [answered]
There is another go project [gojq](https://github.com/itchyny/gojq) which also implements the `jq` syntax and works with yaml files as well as json. How this two projects compare?

**Comment by mikefarah** [comment, marked as answer]
`yq` has been built specifically to handle `yaml` - so it handle multi-docs yaml files, comments, aliases, merge anchors, styling, tags and has operators specifically to manipulate that metadata, and does its best to keep that formatting (though it's not perfect) when editing.

I also noticed that gojq doesn't keep the order of map keys.

I'm sure there are other differences too.

**Comment by kpym** [comment]
@mikefarah Thanks for all these infos. Indeed the main difference seems to be that `gojq` targets primarily json and yaml is the icing on the cake. While `yq` primarily targets `yaml` with all that follows. 

Great project, thanks!
